src/pybind/mgr/dashboard/controllers/_crud.py

```diff
@@ -52,10 +57,18 @@ class TableAction(NamedTuple):
     routerLink: str  # redirect to...
 
 
-class TableComponent(NamedTuple):
-    columns: List[TableColumn] = []
-    columnMode: str = 'flex'
-    toolHeader: bool = True
+class TableComponent(SerializableClass):
+    def __init__(self) -> None:
+        self.columns = []
+        self.columnMode = 'flex'
+        self.toolHeader = True
+
+    def __iter__(self):
+        yield from {
+            "columns": self.columns,
+            "columnMode": self.columnMode,
+            "toolHeader": self.toolHeader,
+        }.items()
```

Comment:
Instead of using a dummy class (`SerializableClass`) and adding some boilerplatish code that could be easily made generic, why not constructing this `__iter__` method from a metaclass implementing the `__new__` method?

By doing that you can keep the abstract class style with class variables (left-side).

[This](https://stackoverflow.com/questions/5434400/python-make-class-iterable) is the simplest/ugliest implementation (I think it could even work without a metaclass, just regular subclassing).

That example can by improved my implementing the `__new__` method, which saves your from filtering out `__` attributes.

Additionally, since what you're doing here is returning "dict-like view" to objects, it might be interesting to comply with the [existing abstract interface](https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes) (not only would it require the `__iter__` method, but the `__contains__` one too).